Build failure: easytag

### Nixpkgs version

- Unstable (26.05)

### Steps to reproduce

build easytag

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/317988400

### Relevant log output

```console
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/zgycy5id7ad7i09mla6vzzd9b9xmibi9-easytag-2.4.3.tar.xz
source root is easytag-2.4.3
setting SOURCE_DATE_EPOCH to timestamp 1480966458 of file "easytag-2.4.3/ChangeLog"
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/nqrp08ydp3gzwxgn4312jqqhlqbq015p-easytag-2.4.3-ogg-corruption.patch?id=b175a159c1138702bdfb009ff4d6565019ed3c4a
patching file src/tags/vcedit.c
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.sub
Updating Autotools / GNU config script to a newer upstream version: ./build-aux/config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
fixing libtool script ./build-aux/ltmain.sh
./configure
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/lw117lsr8d585xs63kx5k233impyrq7q-bash-5.3p3/bin/sh"
configure flags: --disable-static --disable-dependency-tracking --prefix=/nix/store/yxgji52sspxg8m7y6vnilbq9xp9w5ahf-easytag-2.4.3
checking for a BSD-compatible install... /nix/store/d75200gb22v7p0703h5jrkgg8bqydk5q-coreutils-9.8/bin/install -c
checking whether build environment is sane... yes
checking for a thread-safe mkdir -p... /nix/store/d75200gb22v7p0703h5jrkgg8bqydk5q-coreutils-9.8/bin/mkdir -p
checking for gawk... gawk
checking whether make sets $(MAKE)... yes
checking whether make supports nested variables... yes
checking whether to enable maintainer-specific portions of Makefiles... yes
checking whether make supports nested variables... (cached) yes
checking build system type... x86_64-pc-linux-gnu
checking host system type... x86_64-pc-linux-gnu
checking how to print strings... printf
checking for style of include used by make... GNU
checking for gcc... gcc
checking whether the C compiler works... yes
checking for C compiler default output file name... a.out
checking for suffix of executables... 
checking whether we are cross compiling... no
checking for suffix of object files... o
checking whether we are using the GNU C compiler... yes
checking whether gcc accepts -g... yes
checking for gcc option to accept ISO C89... none needed
checking whether gcc understands -c and -o together... yes
checking dependency style of gcc... none
checking for a sed that does not truncate output... /nix/store/k06ssckzrzn9jjvvs4n62m6567zmbx6x-gnused-4.9/bin/sed
checking for grep that handles long lines and -e... /nix/store/qfmq7p42ak5yn389qvx7zpxkan5i4xiy-gnugrep-3.12/bin/grep
checking for egrep... /nix/store/qfmq7p42ak5yn389qvx7zpxkan5i4xiy-gnugrep-3.12/bin/grep -E
checking for fgrep... /nix/store/qfmq7p42ak5yn389qvx7zpxkan5i4xiy-gnugrep-3.12/bin/grep -F
checking for ld used by gcc... ld
checking if the linker (ld) is GNU ld... yes
checking for BSD- or MS-compatible name lister (nm)... nm
checking the name lister (nm) interface... BSD nm
checking whether ln -s works... yes
checking the maximum length of command line arguments... 1572864
checking how to convert x86_64-pc-linux-gnu file names to x86_64-pc-linux-gnu format... func_convert_file_noop
checking how to convert x86_64-pc-linux-gnu file names to toolchain format... func_convert_file_noop
checking for ld option to reload object files... -r
checking for objdump... objdump
checking how to recognize dependent libraries... (cached) pass_all
checking for dlltool... no
checking how to associate runtime and link libraries... printf %s\n
checking for archiver @FILE support... @
checking for strip... strip
checking for ranlib... ranlib
checking command to parse nm output from gcc object... ok
checking for sysroot... no
checking for a working dd... /nix/store/d75200gb22v7p0703h5jrkgg8bqydk5q-coreutils-9.8/bin/dd
checking how to truncate binary pipes... /nix/store/d75200gb22v7p0703h5jrkgg8bqydk5q-coreutils-9.8/bin/dd bs=4096 count=1
checking for mt... no
checking if : is a manifest tool... no
checking how to run the C preprocessor... gcc -E
checking for ANSI C header files... yes
checking for sys/types.h... yes
checking for sys/stat.h... yes
checking for stdlib.h... yes
checking for string.h... yes
checking for memory.h... yes
checking for strings.h... yes
checking for inttypes.h... yes
checking for stdint.h... yes
checking for unistd.h... yes
checking for dlfcn.h... yes
checking for objdir... .libs
checking if gcc supports -fno-rtti -fno-exceptions... no
checking for gcc option to produce PIC... -fPIC -DPIC
checking if gcc PIC flag -fPIC -DPIC works... yes
checking if gcc static flag -static works... no
checking if gcc supports -c -o file.o... yes
checking if gcc supports -c -o file.o... (cached) yes
checking whether the gcc linker (ld -m elf_x86_64) supports shared libraries... /nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: warning: ld-linux-x86-64.so.2, needed by /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6, not found (try using -rpath or -rpath-link)
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: warning: cannot find entry symbol _start; not setting start address
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_audit_symbind_alt@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_catch_exception@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__nptl_change_stack_perm@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_deallocate_tls@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__rseq_size@GLIBC_2.35'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_argv@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_allocate_tls@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_find_dso_for_object@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__libc_enable_secure@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_fatal_printf@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__tunable_is_initialized@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__tunable_get_val@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_signal_exception@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_allocate_tls_init@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_rtld_di_serinfo@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_rtld_global@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__tls_get_addr@GLIBC_2.3'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `__libc_stack_end@GLIBC_2.2.5'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_signal_error@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_dl_audit_preinit@GLIBC_PRIVATE'
/nix/store/cl88v2m1y5q3k6jlkq5jjf73nmfgl1px-binutils-2.44/bin/ld: /nix/store/j193mfi0f921y0kfs8vjc1znnr45ispv-glibc-2.40-66/lib/libc.so.6: undefined reference to `_rtld_global_ro@GLIBC_PRIVATE'
yes
checking whether -lc should be explicitly linked in... no
checking dynamic linker characteristics... GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking whether stripping libraries is possible... yes
checking if libtool supports shared libraries... yes
checking whether to build shared libraries... yes
checking whether to build static libraries... no
checking whether NLS is requested... yes
checking for intltool >= 0.50.0... 0.51.0 found
checking for intltool-update... /nix/store/4rnsqfwfw4iby9k4iz6zs41njkzz2hcl-intltool-0.51.0/bin/intltool-update
checking for intltool-merge... /nix/store/4rnsqfwfw4iby9k4iz6zs41njkzz2hcl-intltool-0.51.0/bin/intltool-merge
checking for intltool-extract... /nix/store/4rnsqfwfw4iby9k4iz6zs41njkzz2hcl-intltool-0.51.0/bin/intltool-extract
checking for xgettext... /nix/store/1dax567dy7cf55x25a1ryg6z6hl3lvzw-gettext-0.25.1/bin/xgettext
checking for msgmerge... /nix/store/1dax567dy7cf55x25a1ryg6z6hl3lvzw-gettext-0.25.1/bin/msgmerge
checking for msgfmt... /nix/store/1dax567dy7cf55x25a1ryg6z6hl3lvzw-gettext-0.25.1/bin/msgfmt
checking for gmsgfmt... /nix/store/1dax567dy7cf55x25a1ryg6z6hl3lvzw-gettext-0.25.1/bin/msgfmt
checking for perl... /nix/store/d22ak2kndm1z2c7fmx3s4lim9w44wk7s-perl-5.42.0/bin/perl
checking for perl >= 5.8.1... 5.42.0
checking for XML::Parser... ok
checking for appstream-util... no
checking for gcc... (cached) gcc
checking whether we are using the GNU C compiler... (cached) yes
checking whether gcc accepts -g... (cached) yes
checking for gcc option to accept ISO C89... (cached) none needed
checking whether gcc understands -c and -o together... (cached) yes
checking dependency style of gcc... (cached) none
checking whether we are using the GNU C++ compiler... yes
checking whether g++ accepts -g... yes
checking dependency style of g++... none
checking how to run the C++ preprocessor... g++ -E
checking for ld used by g++... ld -m elf_x86_64
checking if the linker (ld -m elf_x86_64) is GNU ld... yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking for g++ option to produce PIC... -fPIC -DPIC
checking if g++ PIC flag -fPIC -DPIC works... yes
checking if g++ static flag -static works... no
checking if g++ supports -c -o file.o... yes
checking if g++ supports -c -o file.o... (cached) yes
checking whether the g++ linker (ld -m elf_x86_64) supports shared libraries... yes
checking dynamic linker characteristics... (cached) GNU/Linux ld.so
checking how to hardcode library paths into programs... immediate
checking how to run the C++ preprocessor... g++ -E
checking for a sed that does not truncate output... (cached) /nix/store/k06ssckzrzn9jjvvs4n62m6567zmbx6x-gnused-4.9/bin/sed
checking pkg-config is at least version 0.24... yes
checking for git... ${SHELL} /build/easytag-2.4.3/build-aux/missing git
checking for gdk-pixbuf-pixdata... /nix/store/zsa9rgmanx1l9h7i76gfvv440n90vylg-gdk-pixbuf-2.44.4-dev/bin/gdk-pixbuf-pixdata
checking for glib-mkenums script... /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/bin/glib-mkenums
checking for XML catalog (/etc/xml/catalog)... not found
checking for xmlcatalog... /nix/store/mvfalwl76ms34wxrh9jsw6g46fsl9nbh-libxml2-2.15.1-bin/bin/xmlcatalog
checking for xsltproc... no
checking for DocBook XML DTD V4.4 in XML catalog... not found
checking for DocBook XSL Stylesheets in XML catalog... not found
checking for desktop-file-validate... no
checking for gtester... /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/bin/gtester
checking for gtester-report... /nix/store/j359ndglk11v5j47ra66lc1vxlk2dbjw-glib-2.86.3-dev/bin/gtester-report
checking for itstool... itstool
checking for xmllint... xmllint
checking for yelp... no
configure: WARNING: Yelp is required at runtime for user help
checking whether to build with mingw32 for a Windows host... no
checking for recv in -lsocket... no
checking for library containing ID3Tag_Link... -lid3 -lstdc++
checking the id3lib version... configure: error: could not determine id3lib version
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.18.2, NixOS, 25.11 (Xantusia), 25.11.20251229.89dbf01`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.31.2`
 - nixpkgs: `/nix/store/rv4z0vd7v0pkadinxd1c1xmd4b850ldr-source`

### Notify maintainers

@matteo-pacini 

---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/ea/easytag/package.nix
+++ b/pkgs/by-name/ea/easytag/package.nix
@@ -40,7 +40,12 @@ stdenv.mkDerivation rec {
     })
   ];
 
-  NIX_LDFLAGS = "-lid3tag -lz";
+  env = {
+    NIX_LDFLAGS = "-lid3tag -lz";
+    # Note: this allows id3lib to be found at configure time, and also prevents
+    # compilation errors on Linux (GCC 15). Clang / Darwin seems to be unaffected.
+    NIX_CFLAGS_COMPILE = lib.optionalString stdenv.cc.isGNU "-std=c17 -Wno-implicit-function-declaration";
+  };
 
   nativeBuildInputs = [
     pkg-config

